File: palindrome/palindrome.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution{
    public:
        bool isPalindrome(string s){
            int i = 0, j = s.length() - 1;
            while(i < j){
                if(s[i] == s[j]){
                    i++;
                    j--;
                }else{
                    return false;
                }
            }
            return true;
        }
};

int main(){
    Solution s;
    if(s.isPalindrome("aba")){
        cout << "The string is palindrome" << endl;
    }else{
        cout << "The strng is not palindrome" << endl;
    }
}

File: palindrome/palindrome.py
class Solution:

    def palindrome_rev_string(self, s:str)->bool:
        """
        This function returns true if the string is a palindrome using string reversing.
        """
        return s == s[::-1]
    

    def is_palindrome(seld, s:str)->bool:
        """
        This function uses Two pointer technique to find if a string is palindrome or not.
        """
        i,j = 0, len(s) - 1
        while i < j:
            if s[i] == s[j]:
                i, j = i + 1, j - 1
            else:
                return False
        return True